pkg/controller/certificaterequests/sync.go

```diff
@@ -121,55 +114,40 @@ func (c *Controller) Sync(ctx context.Context, cr *v1alpha1.CertificateRequest)
 		c.recorder.Eventf(crCopy, corev1.EventTypeWarning, "BadConfig", "Resource validation failed: %v", el.ToAggregate())
 
 		apiutil.SetCertificateRequestCondition(crCopy, v1alpha1.CertificateRequestConditionReady,
-			v1alpha1.ConditionFalse, reasonFailed, fmt.Sprintf("Validation failed: %s", el.ToAggregate()))
+			v1alpha1.ConditionFalse, v1alpha1.CertificateRequestReasonFailed, fmt.Sprintf("Validation failed: %s", el.ToAggregate()))
 		return nil
 	}
 
+	defer c.setCertificateRequestStatus(crCopy)
+
```

Comment:
This looks like new code to me?

Reply:
I've merged func (c *Controller) sign(ctx context.Context, cr *v1alpha1.CertificateRequest, issuer issuer.Interface) error {
to just be included into this one